src/status.ts

```diff
@@ -2,258 +2,441 @@ import * as vscode from 'vscode';
+    }
+    return ActiveFolderButton._toolTip;
+  }
+  protected isVisible():boolean {
+    return Boolean(vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1 && !!this.getText());
+  }
+}
+
+class CMakeStatus extends Button {
+  settingsName = 'status';
+  command = "cmake.setVariant";
+  tooltip = localize('click.to.select.variant.tooltip', 'Click to select the current build variant');
+
+  private _buildTypeLabel: string = localize('unconfigured', 'Unconfigured');
   private _statusMessage: string = localize('loading.status', 'Loading...');
-  setStatusMessage(v: string) {
+  set buildTypeLabel(v: string) {
+    this._buildTypeLabel = v;
+    this.update();
+  }
+  set statusMessage(v: string) {
     this._statusMessage = v;
-    this._reloadStatusButton();
+    this.update();
+  }
+  protected getTextNormal() {
+    return `CMake: ${this._buildTypeLabel}: ${this._statusMessage}`;
+  }
+  protected getTextShort() {
+    return `${this._buildTypeLabel}: ${this._statusMessage}`;
   }
+}
+
+class KitSelection extends Button {
+  private static readonly _noActiveKit = localize('no.active.kit', 'No active kit');
+  private static readonly _noKitSelected = localize('no.kit.selected', 'No Kit Selected');
 
-  /**
-   * The name of the currently active target to build
-   */
-  private _targetName: string = '';
-  public get targetName(): string { return this._targetName; }
-  public set targetName(v: string) {
-    this._targetName = v;
-    this._targetButton.text = `[${v}]`;
-    this.reloadVisibility();
+  settingsName = 'kit';
+  command = 'cmake.selectKit';
+  icon = 'tools';
+  tooltip = localize('click.to.change.kit.tooltip', 'Click to change the active kit');
+
+  protected getTextNormal():string {
+    const text = this.text;
+    if (text === SpecialKits.Unspecified) {
+      return KitSelection._noActiveKit;
+    }
+    if (text.length === 0) {
+      return KitSelection._noKitSelected;
+    }
+    return text;
   }
 
-  setLaunchTargetName(v: string) {
-    this._launchTargetNameButton.text = v;
-    this._reloadDebugButton();
+  protected getTextShort() {
+    let len = this._config.statusbar.advanced?.kit?.length || 0;
+    if (!Number.isInteger(len) || len <= 0) {
+      len = 20;
+    }
+    let text = this.getTextNormal();
+    if (len + 3 < text.length) {
+      text = `${text.substr(0, len)}...`;
+    }
+    return text;
+  }
+}
+
+class BuildTargetSelectionButton extends Button {
+  settingsName = 'buildTarget';
+  command = 'cmake.setDefaultTarget';
+  tooltip = localize('set.active.target.tooltip', 'Set the active target to build');
+
+  protected getTextNormal():string {
+    return `[${this.text}]`;
+  }
+}
+class LaunchTargetSelectionButton extends Button {
+  settingsName = 'launchTarget';
+  command = 'cmake.selectLaunchTarget';
+  tooltip = localize('select.target.tooltip', 'Select the target to launch');
+
+  protected getTextNormal():string {
+    return `[${this.text}]`;
```

Comment:
this should show `[-]` if no target is selected.

i can hide this button if no target is selected, but i personally think it's good to have the button. If you just want to select the target without debugging or launching, you'd have to use the commands or the outline.
I thought about adding a text if no target is selected but that takes way too much space.

i think it's a good compromise to have this with as little space usage as possible there.

Reply:
The first time you debug or launch a target it will ask you to select one, so I don't think we need to add this to the status bar when it's unset.  Perhaps you can change the tooltip based on whether a target is selected or not to explain that you will select and debug a target or select and launch a target, but I would rather not add `[-]` to the status bar as it provides no real value.

Reply:
i removed the text.